eventlet/support/greendns.py

```diff
@@ -400,7 +400,8 @@ def end():
                     return end()
 
         # Main query
-        step(self._resolver.query, qname, rdtype, rdclass, tcp, source, raise_on_no_answer=False)
+        step(self._resolver.resolve, qname, rdtype, rdclass, tcp, source,
+             raise_on_no_answer=False, search=True)
```

Comment:
`resolve` and `query` doesn't have the same signature, and `query` call `resolve` with `search=True`, while this option is set to `None` by default.